gdc_ng_models/models/batch.py

```diff
@@ -0,0 +1,131 @@
+import json
+import sqlalchemy
+from sqlalchemy import orm
+from sqlalchemy.ext import declarative
+
+Base = declarative.declarative_base()
+
+
+class Batch(Base):
+    """Batch class to represent a collection of nodes.
+
+    Attributes:
+        id: unique identifier for the batch
+        name: name given to the batch
+        description: optional description for the batch
+        project_id: project the batch is a part of
+        created_datetime: the date and time when the batch is created
+    """
+
+    __tablename__ = "batch"
+    __mapper_args__ = {"eager_defaults": True}
+
+    # start sequence at high number to avoid collisions with existing batch_id
+    id_seq = sqlalchemy.Sequence("batch_id_seq", metadata=Base.metadata, start=1000)
+
+    id = sqlalchemy.Column(
+        sqlalchemy.BigInteger,
+        nullable=False,
+        primary_key=True,
+        server_default=id_seq.next_value(),
+    )
+    name = sqlalchemy.Column(sqlalchemy.Text, nullable=False)
+    description = sqlalchemy.Column(sqlalchemy.Text, nullable=True)
+    project_id = sqlalchemy.Column(sqlalchemy.String(64), nullable=False, index=True)
+    created_datetime = sqlalchemy.Column(
+        sqlalchemy.DateTime(timezone=True),
+        nullable=False,
+        server_default=sqlalchemy.text("now()"),
+    )
+
+    members = orm.relationship(
+        "BatchMembership",
+        back_populates="batch",
+        cascade="all, delete, delete-orphan",
+        passive_deletes=True,
+    )
+
+    def __repr__(self):
+        return "<Batch(id='{}', name='{}', description='{}', project_id='{}', created_datetime='{}')>".format(
+            self.id,
+            self.name,
+            self.description,
+            self.project_id,
+            self.created_datetime.isoformat(),
+        )
+
+    def __eq__(self, other):
```

Comment:
does the declarative base give us equality checking already?

Reply:
sqlalchemy gives us identity checking, it only allows one object with a particular primary key to be loaded into memory per session. i thought it might be nice for users when they needed to check for equality outside the session (comparing all attributes of two objects). i ran into this when i wanted to do checking in some tests i had written, ended up writing a function to do what __eq__ is doing. we can just let the user write it themselves though if we don't want a default implementation

Reply:
perhaps we leave this out until we have a specific use case.  in general, i've seen four types of equality checks with persisted entities:
1) by its primary identifier, which is the `id` in this case.
2) by its natural attributes, which are `name` and `project` in this case.
3) by the entity, which is all the fields on the `batch` table.
4) by the entire logic entity, which is the `batch` row plus its `batch_membership` rows.

Without having a specific use case, it's unclear which we would want to use.  And since there are many options that make sense in difference contexts, we may want separate comparison functions instead of overriding `__eq__`. Using the default identity check for now gives us a consistent behavior with the rest of our models, and allows us to defer the decision about whether or not we want to modify the equality behavior.